connector.go

```diff
@@ -98,13 +100,11 @@ func (c *connector) Connect(ctx context.Context) (driver.Conn, error) {
 	}
 
 	// Enable TCP Keepalives on TCP connections
-	if tc, ok := mc.netConn.(*net.TCPConn); ok {
-		if err := tc.SetKeepAlive(true); err != nil {
-			// Don't send COM_QUIT before handshake.
-			mc.netConn.Close()
-			mc.netConn = nil
-			return nil, err
-		}
+	if err := enableKeepAlive(mc.netConn); err != nil {
+		// Don't send COM_QUIT before handshake.
+		mc.netConn.Close()
+		mc.netConn = nil
+		return nil, err
```

Comment:
I think keepalive shouldn't mandatory.

```suggestion
		c.cfg.Logger.Print(err)
```